src/update.c

```diff
@@ -396,13 +447,23 @@ static int main_update()
 	/* The new subscription is seeded from the list of currently installed bundles
 	 * This calls add_subscriptions which recurses for new includes */
 	grabtime_start(&times, "Add Included Manifests");
+	bool reattempt_download = true;
+add_dependencies:
 	ret = add_included_manifests(server_manifest, current_version, &latest_subs);
 	grabtime_stop(&times);
 	if (ret) {
+		if (reattempt_download) {
+			/* Fall back to using the full manifests */
+			server_manifest->manifests = full_manifests;
+			reattempt_download = false;
+			printf("Cannot load server MoM iterative sub-manifests, trying to download full sub-manifests\n");
```

Comment:
This will only fail if `add_subscriptions` fails entirely right? Not when it can't get a couple iterative submanifests? I just want to make sure we aren't printing this every time we get an iterative manifest we can't use

Reply:
This is one of the parts of the code where I'm still doing some evaluation to see what would be the best approach.

The main issue is that the _to manifests_ are downloaded as part of adding included manifests, so this is even before getting to load server sub-manifests, by the time the server sub-manifests are loaded the manifests were already downloaded.

As the code is right now, if any of the manifests fail to be downloaded while adding the included manifests, then the whole update is aborted (https://github.com/clearlinux/swupd-client/blob/master/src/update.c#L399).

Now, if this happen when using iterative manifests, we no longer want to abort the update but we want to try now with the full manifest. As the code is in this patch, if any of the manifests fail to be downloaded (iterative or full) while adding included manifests, the process of adding included manifests will be re-attempted but this time using only all full manifests (not only the full manifest of the bundle that failed). I'm seeing if there is an alternative for adding this functionality.

But yes, the message about not being able to load the iterative sub-manifests would only be shown once.

Reply:
So the first time it fails to download an iterative manifest it falls back entirely to full manifests? That is not ideal but maybe not a dealbreaker. The only iterative manifests that will be attempted to be downloaded are the iterative manifests from the current version, and not *all* iterative manifests, correct?

Reply:
Correct.

I will continue exploring some options anyway to see if there is a way of retrying only on the failed iterative manifest without being too intrusive with the current code.

Reply:
I agree with matthew on that. It would be better to download only the full manifest of the iterative manifest that has problems, but it isn't a dealbreaker.